src/esperanza/finalizationstate.cpp

```diff
@@ -1120,4 +1120,10 @@ bool FinalizationState::IsFinalizerVoting(const Validator &finalizer) const {
   return IsInDynasty(finalizer, m_current_dynasty) || IsInDynasty(finalizer, m_current_dynasty - 1);
 }
 
+uint32_t FinalizationState::GetCurrentDynastyEpochStart() const {
```

Comment:
This function is too special. I'd write `GetDynastyEpochStart` instead, then user can get current dynasty start by `GetDynastyStartEpoch(GetCurrentDynasty())`.

Reply:
There is no use-case for it. We can always extend it if needed or even return the const reference to the underline map. I'd prefer to keep the interface as simple as possible according to the current needs.

Reply:
It doesn't make interface simple, but makes it limited. It's a utility function, we should think about how it could be used but not how it's used at the moment. After all, it's a purpose of any function decomposition...

Reply:
My point is that it is robust and straightforward and I don't see yet a use case for a more generic interface. I see its usage in the RPC only.

 If we do that, then this function needs to handle the case when the inputted value is out of range. Then we should come up with the case: should we assert here, should we throw an exception, do we need to log it, what is the text of that log should be? And then once we made more generic function, probably it makes sense to cover it with tests to ensure what we implemented it correctly. And all this discussion/effort will be spent for smth that is not needed and once we actually need more generic interface it might be "limited" too because we would need to access the underline map. 

I'd agree to work on a more generic approach if it touches the protocol or smth that is hard to change later.